solver/src/settlement_submission/flashbots_settlement.rs

```diff
@@ -0,0 +1,371 @@
+            Ok(gas_price) => Err(anyhow!(
+                "gas station gas price {} is larger than cap {}",
+                gas_price.cap(),
+                self.gas_price_cap
+            )),
+            Err(err) => Err(err),
+        }
+    }
+
+    /// Keep submitting the settlement transaction to the flashbots network as gas price changes.
+    ///
+    /// Returns when simulation of the transaction fails. This likely happens if the settlement
+    /// becomes invalid due to changing prices or the account's nonce changes.
+    ///
+    /// Potential transaction hashes are communicated back through a shared vector.
+    async fn submit_with_increasing_gas_prices_until_simulation_fails(
+        &self,
+        target_confirm_time: Duration,
+        nonce: U256,
+        settlement: Settlement,
+        gas_estimate: U256,
+        transactions: &mut Vec<H256>,
+    ) -> MethodError {
+        const UPDATE_INTERVAL: Duration = Duration::from_secs(10);
+
+        // The amount of extra gas it costs to include the payment to block.coinbase interaction in
+        // an existing settlement.
+        let gas_estimate = gas_estimate + U256::from(18346);
+        let target_confirm_time = Instant::now() + target_confirm_time;
+
+        // gas price and raw signed transaction
+        let mut previous_tx: Option<(EstimatedGasPrice, String)> = None;
+
+        loop {
+            // get gas price
+            // Account for some buffer in the gas limit in case racing state changes result in slightly more heavy computation at execution time.
+            let gas_limit = gas_estimate.to_f64_lossy() * ESTIMATE_GAS_LIMIT_FACTOR;
+            let time_limit = target_confirm_time.saturating_duration_since(Instant::now());
+            let gas_price = match self.gas_price(gas_limit, time_limit).await {
+                Ok(gas_price) => gas_price,
+                Err(err) => {
+                    tracing::error!("gas estimation failed: {:?}", err);
+                    tokio::time::sleep(UPDATE_INTERVAL).await;
+                    continue;
+                }
+            };
+
+            // create transaction
+
+            let tx_gas_cost_in_ether_wei =
+                U256::from_f64_lossy(gas_price.effective_gas_price()) * gas_estimate;
+            let mut settlement = settlement.clone();
+            settlement
+                .encoder
+                .append_to_execution_plan(block_coinbase::PayBlockCoinbase {
+                    amount: tx_gas_cost_in_ether_wei,
+                });
+            let method = super::retry::settle_method_builder(
+                self.contract,
+                settlement.into(),
+                self.account.clone(),
+            )
+            .nonce(nonce)
+            // Wouldn't work because the function isn't payable.
+            // .value(tx_gas_cost_in_ether_wei)
+            .gas(U256::from_f64_lossy(gas_limit))
+            .gas_price(0.0.into());
```

Comment:
Is this going to work with EIP 1559?

Reply:
Yes, because simulation is done using `eth_call` for which gas_price is optional, and if gas_price is equal to None or 0, base fee should be forced to 0.